tests/test_swagger.py

```diff
@@ -1,74 +1,12 @@
 """Unit tests for Annif REST API / Swagger spec"""
+import schemathesis
+from hypothesis import settings
 
+schema = schemathesis.from_path("annif/openapi/annif.yaml")
 
-def test_swagger_cors(app_client):
```

Comment:
Does schemathesis replace this test, and all other tests below? Or should we also keep these tests?

Also, I think we should rename `test_swagger.py` to something else, maybe `test_openapi.py`, because we're not actually using Swagger (2.0) anymore. I just forgot to rename this in the OpenAPI 3.0 PR #649

Reply:
I added a custom check that is run in addition to the default Schemathesis tests to cover checking CORS support:
https://github.com/NatLibFi/Annif/blob/08bf4ce3afd074c24dfcb924fde08684b77f536e/tests/test_openapi.py#L8-L10


[Schemathesis validates the following on the responses](https://schemathesis.readthedocs.io/en/stable/python.html#how-are-responses-checked):


> - not_a_server_error. The response has 5xx HTTP status;
> - status_code_conformance. The response status is not defined in the API schema;
> - content_type_conformance. The response content type is not defined in the API schema;
> - response_schema_conformance. The response content does not conform to the schema defined for this specific response;
> - response_headers_conformance. The response headers does not contain all defined headers.

If I make methods in `rest.py` to return nonsense strings, Schemathesis tests fail:
```
tests/test_openapi.py::test_api[GET /v1/] FAILED                                                                                                                [ 20%]
tests/test_openapi.py::test_api[GET /v1/projects] FAILED                                                                                                        [ 40%]
tests/test_openapi.py::test_api[GET /v1/projects/{project_id}] FAILED                                                                                           [ 60%]
tests/test_openapi.py::test_api[POST /v1/projects/{project_id}/suggest] FAILED                                                                                  [ 80%]
tests/test_openapi.py::test_api[POST /v1/projects/{project_id}/learn] FAILED                                                                                    [100%]
```


If I make error cases to return code `204` instead of `404/503/400` some tests fail:
```
tests/test_openapi.py::test_api[GET /v1/] PASSED                                                                                                                [ 20%]
tests/test_openapi.py::test_api[GET /v1/projects] PASSED                                                                                                        [ 40%]
tests/test_openapi.py::test_api[GET /v1/projects/{project_id}] FAILED                                                                                           [ 60%]
tests/test_openapi.py::test_api[POST /v1/projects/{project_id}/suggest] FAILED                                                                                  [ 80%]
tests/test_openapi.py::test_api[POST /v1/projects/{project_id}/learn] PASSED                                                                                    [100%]
```
(`/learn` method has 204 in the schema, so testing that is passed.)

Both cases are covered also by `test_rest.py`, so I would not keep the old tests here.